molgenis-semantic-mapper/src/main/java/org/molgenis/data/mapper/service/impl/AlgorithmServiceImpl.java

```diff
@@ -114,14 +116,21 @@ public void autoGenerateAlgorithm(EntityType sourceEntityType, EntityType target
 				Object result = jsMagmaScriptEvaluator.eval(algorithm, entity);
 				derivedValue = convert(result, targetAttribute);
 			}
-			catch (RuntimeException e)
+			catch (NullPointerException npe)
 			{
-				if (e.getMessage() == null)
+				try
 				{
-					return algorithmResult.errorMessage(
-							"Applying an algorithm on a null source value caused an exception. Is the target attribute required?");
+					throw new AlgorithmNullValueException();
+
+				}
+				catch (AlgorithmNullValueException anve)
+				{
+					return algorithmResult.errorMessage(anve.getLocalizedMessage());
```

Comment:
then you needn't throw and catch it, I think
but also: not so sure that this matches the javadoc for the exception. An NPE thrown while executing could be anything, really. Plus calling an algorithm on a Null value should be allowed in general